### PR TITLE
Work around /tmp directory issue on FreeBSD

### PR DESCRIPTION
**Pull request by lidl**

### CODE REVIEW

File: Soundlang.class.php [hunk 1]
```diff
@@ -608,10 +608,24 @@ public function installPackage($package) {
...
-		$tmpdir = sys_get_temp_dir();
-		$pkgdir = $tmpdir . '/' . $package['type'] . '-' . $package['module'] . '-' . $package['language'] . '-' . $package['format'] . '-' . $package['version'] . '/';
+		if (PHP_OS == 'FreeBSD') {
+			// Tragically, the default directory that sys_get_temp_dir()
+			// will return (/tmp) on FreeBSD will be have the sticky bit set.
+			// so the files that are unpacked from the tar file will be in
+			// group 'wheel'.  And when the "rename" call happens later,
+			// it will fail with a permission error.
+			// So, use the modules/_cache directory as the workspace.
+			$tmpdir = $amp_conf['AMPWEBROOT']."/admin/modules/_cache";
```
Comment (tm1000) [inline Soundlang.class.php]: this is bad and not a temp folder. This module needs to use /var/lib/asterisk/ as other modules do.

**Comment by lidl** [comment]
Are you saying that the code should just unpack under the tar file undir ASTVARLIBDIR, and do the renames from there, regardless of platform?

**Comment by tm1000** [comment]
Yes. i hate having these freebsd work arounds in because we end up forgetting about them then do a thing that makes them break. If that makes sense. Just create "/var/lib/asterisk/soundtmp". Then when it's done processing and moving you can just delete the whole folder.

**Comment by lidl** [comment]
OK, I will rework this patch to do that for all cases.

**Comment by lidl** [comment]
Patch has been reworked and tested on my host.

**Pull request merged**